### PR TITLE
Listen to SIGHUP to reload config

### PR DESCRIPTION
**Pull request by moitias**
This PR tries to fix the lingering problem of not being able to configure routes dynamically, not allowing gnatsd to change it's infrastructure on the fly. There is another PR (#52) but @derekcollison raised some security concerns and suggested the 'traditional' UNIX way of doing things - sending a SIGHUP to reload the config file, this PR implements a very basic version of just that. 

I'd be happy to keep working on this but would like some feedback on what is the preferred way to close connections (for example, doing it this way does *not* cancel pending route connections as they are not in server.routes until they succesfully connect - which means it includes connections failing to connect and redialing). 

Note that no other data from config is used to reconfigure gnatsd on the fly, I'm planning to figure those out as well. If you know of any pitfalls related to any of the parameters, I'd be glad to know. Changing authorization details is probably the biggest one, should it just disconnect all connections if they change?

As a bonus, there's also another Windows bug fix included, commit ff47f56 that fixed getInterfaceIps for Windows. This was not obvious before due to it throwing errors before logger is initialized properly (maybe executelogcall should just fmt.printf when log.logger == nil instead of dropping the messages quietly?).

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -144,14 +178,20 @@ func (s *Server) handleSignals() {
...
-		}
+			switch sig {
+			case syscall.SIGHUP:
+				Noticef("Server got SIGHUP, reloading configuration..")
+				s.ReloadConfig()
+			case os.Interrupt:
+				// FIXME, trip running?
+				Noticef("Server Exiting..")
+				os.Exit(0)
+			}
```
Comment (krobertson) [inline server/server.go]: Missing close bracket for the for loop. The switch is closed, but not the for loop. This is causing the syntax error on the tests.

**Comment by moitias** [comment]
Woops, my bad - messy merging..

**Comment by mcqueary** [comment]
I'm giving this a gentle "bump" to see if we can move this PR one direction or the other :) @derekcollison @moitias

**Comment by derekcollison** [comment]
We will code this differently.

**Pull request closed without merging**